### ISSUE TITLE
design should not be allowed to load with duplicate deletions or insertions

### ISSUE THREAD

**Issue by DurhamSmith** [UC-Davis-molecular-computing/scadnano#472]
I am unable to assign my scaffold strand a sequence for the following file sc file (converted to txt for upload)

[ss_connected_tile (copy).txt](https://github.com/UC-Davis-molecular-computing/scadnano/files/5202743/ss_connected_tile.copy.txt)

The sequence I want to use is:
[seq.txt](https://github.com/UC-Davis-molecular-computing/scadnano/files/5202766/seq.txt)

which works when being assigned to this .sc file:
[origami_cross_tile_strong_strong .txt](https://github.com/UC-Davis-molecular-computing/scadnano/files/5202769/origami_cross_tile_strong_strong.txt)
Comparing the files there are not glaring differences.

The error I get is:
![Screenshot from 2020-09-10 17-18-22](https://user-images.githubusercontent.com/15962411/92752937-ab67d800-f389-11ea-981e-5065280ea1d1.png)

**Comment by dave-doty** [comment]
The problem was coming from lines like this (this is line 216) in the `ss_connected_tile.copy.sc` file:

```json
{"helix": 16, "forward": false, "start": 16, "end": 24, "deletions": [19, 19]},
```

There are two deletions listed at the same position, which was causing the error.

I've fixed the bug in scadnano so it doesn't cause an error in this case. It now removes duplicate insertions/deletions if they are present in the file, and reports an error if two insertions have the same offset but different lengths.

But I wonder how it got there at all. It shouldn't be possible to add the same deletion twice in the web interface. But I'm not sure if the Python scripting code checks for duplicates. Was this file created with a Python script?

**Comment by DurhamSmith** [comment]
Thanks for the prompt response.  
I used the web interface. The issue in my workflow might me that I have copied a staple strand which was already paired with the scaffold strand which had a deletion inserted. When pasting this staple strand only one deletion was present. The deletion on the pasted staple but not the scaffold. I then went and manually removed and re-added all the deletions.

Let me know if you have any questions on reproducing this issue.
